bitcoin/src/blockdata/script.rs

```diff
@@ -1222,6 +1239,142 @@ mod test {
+        let key = PublicKey::from_str(compressed_key_str).unwrap();
+        let p2pk = Script::builder().push_key(&key).push_opcode(OP_CHECKSIG).into_script();
+        let actual = p2pk.p2pk_pubkey_bytes().unwrap();
+        assert_eq!(actual.to_vec(), key.to_bytes());
+    }
+
+    #[test]
+    fn test_p2pk_public_key_valid_key_and_valid_script_returns_expected_key() {
+        let key_str = "0411db93e1dcdb8a016b49840f8c53bc1eb68a382e97b1482ecad7b148a6909a5cb2e0eaddfb84ccf9744464f82e160bfa9b8b64f9d4c03f999b8643f656b412a3";
+        let key = PublicKey::from_str(key_str).unwrap();
+        let p2pk = Script::builder().push_key(&key).push_opcode(OP_CHECKSIG).into_script();
+        let actual = p2pk.p2pk_public_key().unwrap();
+        assert_eq!(actual, key);
+    }
+
+    #[test]
+    fn test_p2pk_public_key_no_checksig_returns_none() {
+        let key_str = "0411db93e1dcdb8a016b49840f8c53bc1eb68a382e97b1482ecad7b148a6909a5cb2e0eaddfb84ccf9744464f82e160bfa9b8b64f9d4c03f999b8643f656b412a3";
+        let key = PublicKey::from_str(key_str).unwrap();
+        let no_checksig = Script::builder().push_key(&key).into_script();
+        assert_eq!(no_checksig.p2pk_public_key(), None);
+    }
+
+    #[test]
+    fn test_p2pk_public_key_empty_script_returns_none() {
+        let empty_script = Script::builder().into_script();
+        assert!(empty_script.p2pk_public_key().is_none());
+    }
+
+    #[test]
+    fn test_p2pk_public_key_no_key_returns_none() {
+        let no_push_bytes = Script::builder().push_opcode(OP_CHECKSIG).into_script();
+        assert!(no_push_bytes.p2pk_public_key().is_none());
+    }
+
+    #[test]
+    fn test_p2pk_public_key_different_op_code_returns_none() {
+        let key_str = "0411db93e1dcdb8a016b49840f8c53bc1eb68a382e97b1482ecad7b148a6909a5cb2e0eaddfb84ccf9744464f82e160bfa9b8b64f9d4c03f999b8643f656b412a3";
+        let key = PublicKey::from_str(key_str).unwrap();
+        let different_op_code = Script::builder().push_key(&key).push_opcode(OP_NOP).into_script();
+        assert!(different_op_code.p2pk_public_key().is_none());
+    }
+
+    #[test]
+    fn test_p2pk_public_key_incorrect_size_returns_none() {
+        let malformed_key = "21032e58afe51f9ed8ad3cc7897f634d881fdbe49816429ded8156bebd2ffd1";
+        let malformed_key_script = Script::builder()
+            .push_slice(malformed_key.as_bytes())
+            .push_opcode(OP_CHECKSIG)
+            .into_script();
+        assert!(malformed_key_script.p2pk_public_key().is_none());
+
+    }
+
+    #[test]
+    fn test_p2pk_public_key_invalid_key_returns_none() {
+        let malformed_key = "21032e58afe51f9ed8ad3cc7897f634d881fdbe49816429ded8156bebd2ffd1ux";
+        let invalid_key_script = Script::builder()
+            .push_slice(malformed_key.as_bytes())
+            .push_opcode(OP_CHECKSIG)
+            .into_script();
+        assert!(invalid_key_script.p2pk_public_key().is_none());
+        
```

Comment:
Can remove this line. To help devs catch this sort of mistake we have pre-commit hook. Check out the README under the `Githooks` section for more info.